### PR TITLE
Add deleteTagCommand with with tests

### PR DESCRIPTION
**Pull request by fustilio**
Add deleteTag function
Add deleteTagCommand command
Add User Guide (probably can be improved) (also need to rename things to contact in the future)
Add Test for deleteTagCommand

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -733,59 +733,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |filter contacts by tags |view specific contacts that are assigned with specific tags
 
-|`* * *` |user |add profile pictures to contacts |
+|`* * *` |user |the browser to automatically search for the address of the selected person in Google Maps |
```
Comment (ktnc985) [inline docs/DeveloperGuide.adoc]: I think the changes to the user guide made here can be shifted to another PR

**Review by Kennard123661 — CHANGES_REQUESTED** [review]
Please add SystemTest for your DeleteTag

**Review by Kennard123661 — CHANGES_REQUESTED** [review]
Please delete the dev guide backup files

File: src/main/java/seedu/address/MainApp.java [hunk 2]
```diff
@@ -40,6 +40,7 @@
  */
 public class MainApp extends Application {
 
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/MainApp.java]: remove this newline

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -32,6 +36,9 @@
      */
     void deleteParcel(ReadOnlyParcel target) throws ParcelNotFoundException;
 
+    /** Deletes the given tag from every parcel. */
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/Model.java]: do your javadocs properly for this one

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -56,6 +64,7 @@ public ReadOnlyAddressBook getAddressBook() {
         return addressBook;
     }
 
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/ModelManager.java]: remove this newline

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -136,4 +175,5 @@ public boolean equals(Object obj) {
                 && filteredParcels.equals(other.filteredParcels);
     }
 
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/ModelManager.java]: remove newline

File: src/main/java/seedu/address/logic/parser/DeleteTagCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.DeleteTagCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new DeleteCommand object
```
Comment (Kennard123661) [inline src/main/java/seedu/address/logic/parser/DeleteTagCommandParser.java]: update this comment

File: src/main/java/seedu/address/model/Model.java [hunk 7]
```diff
@@ -4,9 +4,13 @@
 import java.util.function.Predicate;
 
 import javafx.collections.ObservableList;
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/Model.java:7]: delete newline?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 8]
```diff
@@ -13,10 +17,14 @@
 import seedu.address.commons.core.ComponentManager;
 import seedu.address.commons.core.LogsCenter;
 import seedu.address.commons.events.model.AddressBookChangedEvent;
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/ModelManager.java]: delete newline?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 9]
```diff
@@ -3,7 +3,11 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
+
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/ModelManager.java]: delete newline

File: src/main/java/seedu/address/model/tag/UniqueTagList.java [hunk 10]
```diff
@@ -111,7 +111,7 @@ public boolean equals(Object other) {
         assert CollectionUtil.elementsAreUnique(internalList);
         return other == this // short circuit if same object
                 || (other instanceof UniqueTagList // instanceof handles nulls
-                        && this.internalList.equals(((UniqueTagList) other).internalList));
```
Comment (Kennard123661) [inline src/main/java/seedu/address/model/tag/UniqueTagList.java:114]: is this indentation wrong @Kowalski985 ?

File: src/test/java/seedu/address/logic/commands/DeleteTagCommandTest.java [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.parcel.ReadOnlyParcel;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.exceptions.TagNotFoundException;
+
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for ListCommand.
```
Comment (Kennard123661) [inline src/test/java/seedu/address/logic/commands/DeleteTagCommandTest.java:28]: Please separate out your integration command test

**Review by Kennard123661 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/Model.java (hunk 7 above)
Comment (fustilio) [reply src/main/java/seedu/address/model/Model.java:7]: I think this newline is needed, otherwise it would've been caught by checkstyle

File: src/test/java/seedu/address/logic/commands/DeleteTagCommandTest.java (hunk 11 above)
Comment (fustilio) [reply src/test/java/seedu/address/logic/commands/DeleteTagCommandTest.java:28]: For this, I think it's better to keep it inside.

**Review by Kennard123661 — APPROVED** [review]
wait for @Kowalski985  to check and merge

**Review by ktnc985 — APPROVED** [review]
